src/streaming/zarr.stream.cpp

```diff
@@ -0,0 +1,494 @@
+#include "macros.hh"
+#include "zarr.stream.hh"
+#include "acquire.zarr.h"
+
+#include <blosc.h>
+
+#include <filesystem>
+
+namespace fs = std::filesystem;
+
+namespace {
+bool
+is_s3_acquisition(const struct ZarrStreamSettings_s* settings)
+{
+    return nullptr != settings->s3_settings;
+}
+
+bool
+is_compressed_acquisition(const struct ZarrStreamSettings_s* settings)
+{
+    return nullptr != settings->compression_settings;
+}
+
+[[nodiscard]]
+std::string
+trim(const char* s)
```

Comment:
This utility function, unrelated to zarr streaming, can be moved to a "utilities" header file. It can be used to clean up input in various parts of the codebase.

Reply:
It's not really used anywhere else, but I can move it to zarr.common.hh/cpp in the next PR